### PR TITLE
Fixed #28584 -- Dropped support for SQLite < 3.7.15.

### PR DESCRIPTION
**Pull request by timgraham**
https://code.djangoproject.com/ticket/28584

### CODE REVIEW

File: django/db/backends/sqlite3/features.py [hunk 1]
```diff
@@ -30,13 +30,9 @@ class DatabaseFeatures(BaseDatabaseFeatures):
...
     supports_cast_with_precision = False
-    uses_savepoints = Database.sqlite_version_info >= (3, 6, 8)
-    supports_index_column_ordering = Database.sqlite_version_info >= (3, 3, 0)
+    uses_savepoints = True
     can_release_savepoints = uses_savepoints
-    can_share_in_memory_db = (
-        Database.__name__ == 'sqlite3.dbapi2' and
-        Database.sqlite_version_info >= (3, 7, 13)
-    )
+    can_share_in_memory_db = Database.__name__ == 'sqlite3.dbapi2'
```
Comment (timgraham) [inline django/db/backends/sqlite3/features.py]: The `Database.__name__ == 'sqlite3.dbapi2'` condition is from 8c99b7920e8187f98cf4d7dbd9918bd6c6da1238. I'm not sure if it's still needed (when would it return `False`?) \cc @coldmind

**Comment by bcail** [comment]
Looks like RHEL 7 ships with sqlite 3.7.17 - is it worth supporting the current version of RHEL?

**Comment by timgraham** [comment]
That version is unaffected by the removed code. The commit message could instead say "< 3.7.15" since that's the maximum version for which a workaround is removed.

**Comment by bcail** [comment]
Thanks. Would it be worth updating the gis doc to say that SQLite 3.7.17+ is supported, instead of 3.8+? Otherwise people on RHEL 7 might think the gis functionality isn't supported for them...

**Comment by ngnpope** [comment]
RHEL 7 doesn't ship with Python 3 AFAICT and Python 2 support was removed in Django 2.0. Also Django 2.1 will require Python 3.5 or later. So surely if you need a custom build of Python 3.5 anyway, then you may as well need a custom build of sqlite 3.8? (Granted, the commit message could be changed, but surely this is splitting hairs.)

**Comment by ngnpope** [comment]
Ok. So not being a RHEL user, I've seen that there is the EPEL, but the point still stands as it only provides Python 3.4, not 3.5 or later.

**Comment by bcail** [comment]
There's also Software Collections, which [provide python 3.5 & 3.6](https://www.softwarecollections.org/en/scls/?search=python&policy=&repo=&order_by=-download_count&per_page=10).

But, it seems like in this case it would needlessly require the extra work, since SQLite 3.7.17 should work just fine. No point making users think they need to install a new version when they don't, right?

**Comment by ngnpope** [comment]
Nothing like providing everything in one place... Fair enough.

**Pull request merged**